src/transformers/modeling_bart.py

```diff
@@ -535,16 +534,18 @@ def forward(
         positions = self.embed_positions(input_ids, use_cache=use_cache)
 
         if use_cache:
-            assert input_ids.shape[1] == 1 or decoder_past_key_values is not None, "pass decoder_past_key_value_states to use_cache"
+            assert (
+                input_ids.shape[1] == 1 or decoder_past_key_values is not None
+            ), "pass decoder_past_key_value_states to use_cache"
             input_ids = input_ids[:, -1:]
             positions = positions[:, -1:]  # happens after we embed them
             # assert input_ids.ne(self.padding_idx).any()
 
         x = self.embed_tokens(input_ids) * self.embed_scale
-        if self.variant == 'xlm':
+        if self.variant == "xlm":
             x = self.layernorm_embedding(x)
         x += positions
-        if self.variant == 'bart':
+        if self.variant == "bart":
```

Comment:
Are `xlm` and `bart` variant work in the same way in `EncoderLayer` and `DecoderLayer`? In both class you're using `self.variant != prelayernorm` and then in the Decoder you used `xlm` and `bart`

Reply:
No, they are intentionally different in the decoder layer!

Reply:
Not a big fan of these `self.variant` parameters to be honest

Reply:
But we really need such a parameters  for blenderbot

Reply:
I feel like we are squeezing more and more models into the BART architecture to make other models work with the Bart architecture -> don't really think this is the right direction. Guess, it's ok for now. But as already discussed with @sshleifer we might need to find a clear direction where Seq2Seq is going in general. Also cc @LysandreJik and @sgugger here